llmfoundry/models/hf/hf_causal_lm.py

```diff
@@ -76,108 +87,106 @@ def __init__(self, om_model_config: _om_model_config_type,
+                )
+            with init_empty_weights(include_buffers=False):
+                model = AutoModelForCausalLM.from_config(
+                    config,
+                    trust_remote_code=trust_remote_code,
                 )
-
-            signal_file_path = '.local_rank0_completed_autoresume'
-            if dist.get_local_rank() == 0:
-                with open(signal_file_path, 'wb') as f:
-                    f.write(b'local_rank0_completed_download')
-
-            # Avoid the collective call until the local rank zero has finished trying to download the checkpoint
-            # so that we don't timeout for large downloads. This syncs all processes on the node
-            with dist.local_rank_zero_download_and_wait(signal_file_path):
-                # Then, wait to ensure every node has finished downloading the checkpoint
-                dist.barrier()
-
-            if dist.get_local_rank() == 0:
-                os.remove(signal_file_path)
-
-            z_loss = om_model_config.get('z_loss', 0.0)
-
-        # elif the model is either a PeftModel or a PreTrainedModel
-        elif isinstance(om_model_config, model_types):
-            model = om_model_config
-            init_device = 'cpu'
-            z_loss = 0.0
-
-        # else, unsupported type
         else:
             raise ValueError(
-                f'om_model_config must be either a DictConfig, PeftModel, or PreTrainedModel, but got {type(om_model_config)}'
-            )
+                f'init_device="{init_device}" must be either "cpu" or "meta".')
+
+        signal_file_path = '.local_rank0_completed_autoresume'
+        if dist.get_local_rank() == 0:
+            with open(signal_file_path, 'wb') as f:
+                f.write(b'local_rank0_completed_download')
+
+        # Avoid the collective call until the local rank zero has finished trying to download the checkpoint
+        # so that we don't timeout for large downloads. This syncs all processes on the node
+        with dist.local_rank_zero_download_and_wait(signal_file_path):
+            # Then, wait to ensure every node has finished downloading the checkpoint
+            dist.barrier()
+
+        if dist.get_local_rank() == 0:
+            os.remove(signal_file_path)
+
+        z_loss = om_model_config.get('z_loss', 0.0)
+
+        # if om_model_config includes lora and peft is installed, add lora modules
+        lora_cfg = om_model_config.get('lora', None)
+        if lora_cfg is not None:
+            if _peft_installed == True:
+                print('Building Lora config...')
+                lora_cfg = LoraConfig(**lora_cfg.args)
+                print('Lora config built.')
+                print('Adding Lora modules...')
+                model = get_peft_model(model, lora_cfg)
+                print('Lora modules added.')
+                print_trainable_parameters(model)
+            else:
+                warn(
```

Comment:
I'd vote for this to be an error rather than a warning.

Reply:
me too